src/redux/auth/authSlice.js

```diff
@@ -0,0 +1,59 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { loginUser, signupUser, logoutUser } from './authThunk';
+
+const initialState = {
+  user: null,
+  token: null,
+  loading: false,
+  error: null,
+};
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder
+
+      .addCase(loginUser.pending, (state) => {
+        state.loading = true;
+        state.error = null;
+      })
+      .addCase(loginUser.fulfilled, (state, action) => {
+        state.loading = false;
+        state.user = action.payload;
+      })
+      .addCase(loginUser.rejected, (state, action) => {
+        state.loading = false;
+        state.error = action.payload;
+      })
+
+      .addCase(signupUser.pending, (state) => {
+        state.loading = true;
+        state.error = null;
+      })
+      .addCase(signupUser.fulfilled, (state, action) => {
+        state.loading = false;
+        state.user = action.payload;
+      })
+      .addCase(signupUser.rejected, (state, action) => {
```

Comment:
## Optional 
Great work implementing the reducer functions. However, even though Redux toolkit allows mutating state, the Eslint configurations does not allow mutating states directly so it may cause issues later in the project with linter errors. Instead you can change the state by returning a new object with the changed attributes, for example:

`.addCase(signupUser.fulfilled, (state, action) => {
       return {
        ...state,
        state.loading = false
       }
      })`